components/AppNav.tsx

```diff
@@ -99,19 +111,27 @@ const UnAuthButton = () => (
   </div>
 )
 
-const AppNav: React.FC<GetAppProps> = ({ data: { loading, session } }) => {
+const AppNav: React.FC<{}> = () => {
+  const router = useRouter()
+  const [session, setSession] = useState<Session>({ lessonStatus: [] })
+  const { data } = useGetAppQuery()
+  useEffect(() => {
+    if (data && data.session) {
+      setSession(data.session)
+    }
+  }, [data])
   const renderButtons = () => {
-    if (!session || _.get(session, 'user.username') === null)
+    if (!session || _.get(session, 'user.username', null) === null) {
       return <UnAuthButton />
+    }
 
     const initial = ''
     // TODO: replace with typing
     const username = _.get(session, 'user.username', '')
 
-    return <AuthButton username={username} initial={initial} />
+    return <AuthButton username={username} initial={initial} router={router} />
```

Comment:
Why pass the router as props? There's no problem in using the `useRouter` hook in more than one component.